### PR TITLE
Bug 1000464 - use WifiContext to register events r=Arthur

### PR DESCRIPTION
**Pull request by EragonJ**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b83463ce2596f6290b377229ff2814833fcbc029

File: apps/settings/js/panels/wifi_status/panel.js [hunk 1]
```diff
@@ -28,10 +29,13 @@ define(function(require) {
...
           elements.security.setAttribute('data-l10n-id', 'securityNone');
         }
-        wifiManager.onconnectioninfoupdate = this._updateNetworkInfo;
+
+        WifiContext.addEventListener('wifiConnectionInfoUpdate',
+          this._updateNetworkInfo);
       },
       onBeforeHide: function() {
-        wifiManager.onconnectioninfoupdate = null;
+        WifiContext.removeEventListener('_wifiConnectionInfoUpdate',
```
Comment (crh0716) [inline apps/settings/js/panels/wifi_status/panel.js]: `_wifiConnectionInfoUpdate`?

File: apps/settings/js/panels/wifi_status/panel.js [hunk 2]
```diff
@@ -28,10 +29,13 @@ define(function(require) {
         } else {
           elements.security.setAttribute('data-l10n-id', 'securityNone');
         }
-        wifiManager.onconnectioninfoupdate = this._updateNetworkInfo;
+
+        WifiContext.addEventListener('wifiConnectionInfoUpdate',
```
Comment (crh0716) [inline apps/settings/js/panels/wifi_status/panel.js]: so sad....

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f444e9c8f40c599a0a10a4d7d3b6bbca055b810c

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#dkr9KKLURgS1PymH8lcRpA

The pull request failed to pass integration tests. It could not be landed, please try again.

**Pull request merged**